src/renderer/component/publishForm/view.jsx

```diff
@@ -73,6 +77,10 @@ class PublishForm extends React.PureComponent<Props> {
     (this: any).getNewUri = this.getNewUri.bind(this);
   }
 
+  componentWillMount() {
```

Comment:
This method is being deprecated, it's good to start using `getDerivedStateFromProps()`

https://reactjs.org/docs/react-component.html#static-getderivedstatefromprops